quantum/rgb_matrix/rgb_matrix.c

```diff
@@ -164,6 +164,15 @@ void eeconfig_debug_rgb_matrix(void) {
     dprintf("rgb_matrix_config.flags = %d\n", rgb_matrix_config.flags);
 }
 
+void rgb_matrix_reload_from_eeprom(void) {
+    /* Reset back to what we have in eeprom */
```

Comment:
After some more looking at the mode/enabled state switching, apparently `rgb_matrix_disable_noeeprom()` needs to be called before reloading the settings from EEPROM:
```suggestion
    rgb_matrix_disable_noeeprom();
    /* Reset back to what we have in eeprom */
```
This should handle various possibilities of the `rgb_matrix_config.enable` state changes:
- If the RGB Matrix was enabled before calling this function:
    - If the RGB Matrix was disabled in the EEPROM state, the added `rgb_matrix_disable_noeeprom()` call will stop the previously running effect and blank all LEDs.
    - If the RGB Matrix was also enabled in the EEPROM state, the added `rgb_matrix_disable_noeeprom()` call will not change anything (it will set `rgb_task_state = STARTING`, but then the subsequent `rgb_matrix_mode_noeeprom()` call will just do the same thing again).
- If the RGB Matrix was disabled before calling this function, the added `rgb_matrix_disable_noeeprom()` call will not do anything, then, if the RGB Matrix was enabled in the EEPROM state, `rgb_matrix_mode_noeeprom()` will start the effect.

So adding that `rgb_matrix_disable_noeeprom()` should fix the case when the RGB Matrix state changes from enabled to disabled without breaking any other cases.

Reply:
Makes a ton of sense. Added